dev/com.ibm.ws.jdbc_fat_derby/test-applications/jdbcapp/src/jdbc/fat/derby/web/JDBCDerbyServlet.java

```diff
@@ -10,37 +10,244 @@
+import javax.naming.InitialContext;
 import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
 import javax.sql.DataSource;
+import javax.transaction.UserTransaction;
 
 import org.junit.Test;
 
+import componenttest.annotation.ExpectedFFDC;
 import componenttest.app.FATServlet;
 
 @SuppressWarnings("serial")
 @WebServlet("/JDBCDerbyServlet")
 public class JDBCDerbyServlet extends FATServlet {
 
-    @Resource(name = "jdbc/ds1", shareable = false, authenticationType = AuthenticationType.APPLICATION)
-    DataSource ds1;
+    @Resource(name = "jdbc/dsfat0ref", lookup = "jdbc/dsfat0")
+    DataSource ds0; //DSConfig (TRAN_SERIALIZABLE) + Res-ref (TRAN_NONE) + Normal JDBC Driver
+
+    @Resource(lookup = "jdbc/dsfat1")
+    DataSource ds1; //DSConfig (TRAN_NONE) + No Res-ref + TRAN_NONE JDBC Driver
+
+    @Resource(name = "jdbc/dsfat2ref", lookup = "jdbc/dsfat2")
+    DataSource ds2; //DSConfig (no iso lvl) + Res-ref (TRAN_NONE) + Normal JDBC Driver
+
+    @Resource(name = "jdbc/dsfat3ref", lookup = "jdbc/dsfat3")
+    DataSource ds3; //DSConfig (TRAN_NONE) + Res-ref (TRAN_NONE) + TRAN_NONE JDBC Driver
+
+    @Resource
+    private UserTransaction tran;
+
+    /**
+     * Create the default table used by the tests.
+     */
+    private void createTable(DataSource ds) throws SQLException {
+        Connection con = ds.getConnection();
+        try {
+            Statement st = con.createStatement();
+            try {
+                st.executeUpdate("drop table cities");
+            } catch (SQLException x) {
+            }
+            String dbProductName = con.getMetaData().getDatabaseProductName().toUpperCase();
+            if (dbProductName.contains("IDS") || dbProductName.contains("INFORMIX")) // Informix JCC and JDBC
+                st.executeUpdate("create table cities (name varchar(50) not null primary key, population int, county varchar(30)) LOCK MODE ROW");
+            else
+                st.executeUpdate("create table cities (name varchar(50) not null primary key, population int, county varchar(30))");
+        } finally {
+            con.close();
+        }
+    }
+
+    /**
+     * Data Source - ds1 = DSConfig(TRAN_NONE) + No Res-ref + TRAN_NONE JDBC Driver
+     *
+     * Ensure that if we try to change Transaction Isolation that we throw an exception.
+     */
+    @Test
+    @ExpectedFFDC({ "java.sql.SQLException" })
+    public void testRejectIsolationChange() throws Throwable {
+        Connection con = null;
+        try {
+            con = ds1.getConnection();
+            con.setTransactionIsolation(Connection.TRANSACTION_READ_COMMITTED);
+            fail("Exception should have been thrown when swiching isolation levels on a driver configured with TRANSACTION_NONE.");
+        } catch (SQLException sql) {
+            assertTrue("Exception message should have contained", sql.getMessage().contains("DSRA4011E"));
```

Comment:
Although we have the FFDC, on a subsequent PR update this to rethrow so that we have the stack more easily for debugging.